### PR TITLE
Clear user before load

### PR DESCRIPTION
**Pull request by jenessatan**
- Clear previous profile before loading currently selected profile
- refactor userprofilepage to allow for navigation between user profiles through back button, profile button, etc.

### CODE REVIEW

File: src/components/pages/UserProfile.js [hunk 1]
```diff
@@ -34,53 +33,49 @@ const useStyles = () => ({
...
+          loadUserDetails({
+            userId,
+            currentUserId: currentUser ? currentUser._id : null,
+          })
+        );
+        if (response !== "success") {
+          history.push("/UserNotFound");
+        }
+      } catch (error) {
+        console.log("Failed to load User Profile page");
```
Comment (abid-salahi) [inline src/components/pages/UserProfile.js:60]: Might not be a good idea to use console.logs in production code for our submission. Better to use snackbar to displayError instead.

Comment (jacquelineyin) [reply src/components/pages/UserProfile.js:60]: the error isn't meant for frontend to display. It's rather a help for developers to see what's happening ;)

But I see your point about production. We can do a final pass on the code before doing documentation to remove all the console logs?

**Review by jacquelineyin — APPROVED** [review]

**Comment by abid-salahi** [comment]
Yeah I suppose. It's just i suspect having console.logs in production code is considered unclean code to the graders. Lets double check on that.

**Comment by jenessatan** [comment]
Good point. We are cleaning up all the console logs right now but we'll put it on a different PR

**Pull request merged**